### DISCUSSION TITLE
Taint tracking Graph chart ?

### DISCUSSION THREAD

**Discussion by Sim4n6** [github/codeql#14747] [Q&A] [answered]
For debugging reasons, what are the ways to visualize the taint tracking graph (edges/nodes)? maybe using a dot graph ? thx in advance

**Comment by owen-mc** [comment]
This isn't answering your question, but in case you aren't familiar with it I thought I should mention that the most common way of debugging taint tracking problems is to [use partial flow](https://codeql.github.com/docs/writing-codeql-queries/debugging-data-flow-queries-using-partial-flow/).

**Comment by aeisenberg** [comment, marked as answer]
The CodeQL CLI can output in dot file format.  See the help in `codeql datbase analyze --help`. I think this is what you're asking for. When running the command, use the `--format=dot`. The results can be viewed by any Graphviz viewer.

The caveat is that the query you are running needs to be `@kind graph` and have a specific format. See https://github.com/github/codeql/discussions/7437 for more info. This is an internal feature, so it's not well documented.

When using the feature be careful that the graph you are creating is small since it's easy to generate something that is too large to visualize. Usually, we try to limit the scope to a single file or function.

**Reply by Sim4n6** [reply]
Perfect, Thanks
